montecarlo/cfast_mcarlo.py

```diff
@@ -514,6 +522,30 @@ def _locate_in_the_middle(self, fire_room):
 
         return location
 
+    def _locate_randomly(self, fire_room):
+        location = {}
+        # locate fire in compartment
+        compa = self.s.query("SELECT * FROM aamks_geom WHERE name=?", (fire_room, ))[0]
+
+        # draw location (uniform across the compartment)
+        dx = uniform(0, compa['width'])
+        dy = uniform(0, compa['depth'])
+
+        x = int(compa['x0']+dx)
+        y = int(compa['y0']+dy)
+        self._fire_height = round((compa['height'] * 0.9 * (1-math.log10(uniform(1,9)))) / 100, 2)
```

Comment:
I understand that we want to locate the fire a little higher than the floor level? Because there is also an option for the fire to always be at floor level

Reply:
No, the idea is to place fire from 0 to 0.9 compartment height, with possibility of further narrowing those ranges in case of CFAST errors.

Maybe I will just adjust parameters of uniform function to change the ranges of fire elevation in only one place?

Reply:
fixed in https://github.com/aamks/aamks/pull/149/commits/1940d97779528233ad4d7643c9ae67204337fb3b